cloudant-sync-datastore-core/src/main/java/com/cloudant/mazha/CouchClient.java

```diff
@@ -461,13 +461,27 @@ public void putAttachmentStream(String id, String rev, String attachmentName, St
         // merge results back in, so there is one list of DocumentRevs per id
         for (BulkGetResponse.Result result : response.results) {
             for (BulkGetResponse.Doc doc : result.docs) {
-                if (doc.ok != null) {
+                if (doc.ok != null){
                     String id = doc.ok.getId();
                     if (!revsMap.containsKey(id)) {
                         revsMap.put(id, new ArrayList<DocumentRevs>());
                     }
                     revsMap.get(id).add(doc.ok);
+                } else {
+                    // error will be set here, so lets just access the map to get at the id
+                    @SuppressWarnings("unchecked")
+                    String id = (String)doc.error.get("id");
```

Comment:
I could, but I'd rather not, since we only need two properties out of the map, and we only need it here, I don't think there is much point to add a whole class to avoid two casts and a localised suppression.